cdap-security/src/main/java/co/cask/cdap/security/authorization/AuthorizerInstantiatorService.java

```diff
@@ -0,0 +1,281 @@
+    return authorizer;
+  }
+
+  /**
+   * Creates a new instance of the configured {@link Authorizer} extension, based on the provided extension jar file.
+   *
+   * @return a new instance of the configured {@link Authorizer} extension
+   */
+  public Authorizer createAuthorizerInstance(File authorizerExtensionJar)
+    throws IOException, InvalidAuthorizerException {
+    Class<? extends Authorizer> authorizerClass = loadAuthorizerClass(authorizerExtensionJar);
+    // Set the context class loader to the AuthorizerClassLoader before creating a new instance of the extension,
+    // so all classes required in this process are created from the AuthorizerClassLoader.
+    ClassLoader oldClassLoader = Thread.currentThread().getContextClassLoader();
+    Thread.currentThread().setContextClassLoader(authorizerClassLoader);
+    LOG.debug("Setting context classloader to {}. Old classloader was {}.", authorizerClassLoader, oldClassLoader);
+    try {
+      Constructor<? extends Authorizer> constructor;
+      try {
+        constructor = authorizerClass.getDeclaredConstructor(Properties.class);
+      } catch (NoSuchMethodException e) {
+        throw new InvalidAuthorizerException(
+          String.format("No suitable constructor for Authorizer extension %s. Please make sure that the extension is " +
+                          "a public class with a public constructor that accepts a single parameter of type %s.",
+                        authorizerClass.getName(), Properties.class.getName()), e);
+      }
+      try {
+        return constructor.newInstance(createExtensionProperties());
+      } catch (InstantiationException | IllegalAccessException | InvocationTargetException e) {
+        throw new InvalidAuthorizerException(
+          String.format("Error while instantiating for authorizer extension %s. Please make sure that the extension " +
+                          "is a public class with a public constructor that accepts a single parameter of type %s.",
+                        authorizerClass.getName(), Properties.class.getName()), e);
+      }
+    } finally {
+      // After the process of creation of a new instance has completed (success or failure), reset the context
+      // classloader back to the original class loader.
+      Thread.currentThread().setContextClassLoader(oldClassLoader);
+      LOG.debug("Resetting context classloader to {} from {}.", oldClassLoader, authorizerClassLoader);
+    }
+  }
+
+  private Properties createExtensionProperties() {
+    Properties extensionProperties = new Properties();
+    for (Map.Entry<String, String> entry :
+      cConf.getValByRegex("^" + Constants.Security.Authorization.AUTHORIZER_EXTENSION_CONFIG_PREFIX).entrySet()) {
+      // remove the prefix
+      String key =
+        entry.getKey().substring(Constants.Security.Authorization.AUTHORIZER_EXTENSION_CONFIG_PREFIX.length());
+      // trim the leading periods
+      if (key.startsWith(".")) {
```

Comment:
This was for the situation where people may do ``Constants.Security.Authorization.AUTHORIZER_EXTENSION_CONFIG_PREFIX + ".my.property.name`` (multiple periods between the prefix and the actual extension property). I made that mistake in a unit test, so added a check. But since the constant won't be available to extension code, it may be an overkill. removed.